.github/workflows/pr-percy-create-screenshots.yml

```diff
@@ -0,0 +1,145 @@
+            echo "Previous workflow run was not successful."
+            exit 1
+          else
+            echo "Previous workflow run was successful."
+          fi
+      - name: Checkout SCM
+        uses: actions/checkout@v4
+
+      - name: Download artifact from workflow run
+        uses: actions/download-artifact@v4
+        with:
+          name: "web-artifact-${{ github.event.workflow_run.id }}"
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          repository: ${{ github.event.workflow_run.repository.full_name }}
+          run-id: ${{ github.event.workflow_run.id }}
+
+      - name: Replace SCM files with artifact files
+        run: |
+          set -e
+          rm -rf templates/docs/examples/ scss/
+          tar xzf build.tar.gz
+          rm build.tar.gz
+          mv examples/ templates/docs/.
+
+      - name: Install Python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: 3.10.14
+
+      - name: Install NodeJS
+        uses: actions/setup-node@v4.0.2
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        run: yarn && pip3 install -r requirements.txt
+
+      - name: Build
+        run: yarn build
+
+      - name: Start testing server
+        env:
+          SECRET_KEY: local_development_fake_key
+        run: ./entrypoint 0.0.0.0:8101 &
+
+      - name: Wait for server availability
+        run: |
+          set -e
+          sleep_interval_secs=2
+          max_wait_time_secs=30
+          wait_time_secs=0
+          while [ "$wait_time_secs" -lt "$max_wait_time_secs" ]; do
+            if curl -s localhost:8101/_status/check -I; then
+              echo "Server is up!"
+              break
+            else
+              wait_time_secs=$((wait_time_secs + sleep_interval_secs))
+              if [ "$wait_time_secs" -ge "$max_wait_time_secs" ]; then
+                echo "[TIMEOUT ERROR]: Local testing server failed to respond within $max_wait_time_secs seconds!"
+                exit 1
+              else
+                echo "Waiting for server to start..."
+                sleep "$sleep_interval_secs"
+              fi
+            fi
+          done
+
+      - name: Get PR data
+        if: github.event.workflow_run.event=='pull_request'
+        id: get_pr_data
+        run: |
+          set -e
+          echo "sha=$(cat pr_head_sha.txt)" >> $GITHUB_OUTPUT
+          echo "pr=$(cat pr_num.txt)" >> $GITHUB_OUTPUT
+
+      - name: Take snapshots & upload to Percy
+        id: percy_snapshot
+        env:
+          PERCY_TOKEN: ${{ secrets.PERCY_TOKEN_WRITE }}
+          # This identifies the diff in the percy diffs list
+          PERCY_BRANCH: "pull/${{ steps.get_pr_data.outputs.pr }}"
+          #PERCY_BRANCH: "${{ github.event.workflow_run.head_repository.full_name }}/${{ github.event.workflow_run.head_branch }}"
+          #PERCY_PULL_REQUEST: ${{ steps.get_pr_data.outputs.pr }}
```

Comment:
@bartaz Yes, I ran into a bug while I was working on #5092 where I noticed that all Percy builds were becoming the baseline because they defaulted to being "main" branch without me specifying `PERCY_BRANCH`.

This sets the diff branch name to a PR identifier, making sure that builds don't become the baseline unless they are approved.

The commented out code was doing the following:

- `PERCY_BRANCH` - I was trying to set the branch name to `<fork_repo_username>/<fork_repo_name>/<fork_pr_head_branch_name>` in an attempt to make the diff branches unique. However I simplified this to the `pull/` format as that is what is currently used in the project.
- `PERCY_PULL_REQUEST` - links the diff to a specific PR by number. I think this requires the Percy->GH integration to be enabled, but I don't have permissions to enable it in the Percy portal.

See more about Percy CLI environment vars [here](https://www.browserstack.com/docs/percy/get-started/set-env-var).

I'll remove these in a commit shortly.